pkg/network/tracer/connection/ebpf_tracer.go

```diff
@@ -314,12 +314,15 @@ func (t *ebpfTracer) Start(callback func(*network.ConnectionStats)) (err error)
 		return fmt.Errorf("error initializing port binding maps: %s", err)
 	}
 
+	t.closeConsumer.Start(callback)
+	t.failedConnConsumer.Start()
+
 	if err := t.m.Start(); err != nil {
+		t.closeConsumer.Stop()
+		t.failedConnConsumer.Stop()
```

Comment:
worth adding inline documentation here, someone might change it in the future without understanding why you've made this change